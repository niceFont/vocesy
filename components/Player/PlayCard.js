import { Card } from "react-bootstrap";
import Renderer from "../../components/Editor/Renderer"


const PlayCard = (props) => {

    return (
					<Card
						style={{
							position: "absolute",
							height: "22em",
							width: "16em",
						}}
					>
						<Card.Header>Front</Card.Header>
						<Card.Body className="text-center">
							<Renderer data={props.data}></Renderer>
						</Card.Body>
					</Card>
    )
}


export default PlayCard