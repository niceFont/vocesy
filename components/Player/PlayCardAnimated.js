import { useSpring, animated } from "react-spring"
import { Card, Container } from "react-bootstrap"
import React, {useState} from "react"
import Renderer from "../Editor/Renderer";
import whyDidYouRerender from "@welldone-software/why-did-you-render";

whyDidYouRerender(React)

const PlayCardAnimated = (props) => {
	
	const [flipped, toggleFlipped] = useState(false)
	const { transform, opacity } = useSpring({
		opacity: flipped ? 1 : 0,
		transform: `perspective(600px) rotateY(${flipped ? 180 : 0}deg)`,
		config: {
			mass: 10, tension: 500, friction: 80, duration: 200
		}
	})


	return (
		
		
		<Container id="toggler" style={{
			overflow: "hidden",
			margin: "25px 0 25px 0",
			position: "absolute",
			cursor: "pointer"
		}} onClick={() => {
			toggleFlipped(o => !o)
		}}>
			<div>

				<animated.div id="front" className="justify-content-center text-center" style={{
					opacity: opacity.interpolate(o => 1 - o), transform,
					display: "flex",
					justifyContent: "center"
				}} >
					<Card className="mx-auto" style={{
						height: "22rem",
						width: "16em",
						position: "absolute"
					}}>
						<Card.Header>
							<Card.Subtitle>Front</Card.Subtitle>
						</Card.Header>
						<Card.Body style={{ height: "15em" }}>
							<Renderer data={props.data.front}></Renderer>
						</Card.Body>
					</Card>
				</animated.div>
				<animated.div id="back" style={{
					opacity, transform: transform.interpolate(t => `${t} rotateY(180deg)`),
					display: "flex",
					justifyContent: "center"
				}}>
					<Card style={{
						height: "22rem",
						width: "16em",
					}}>
						<Card.Header>
							<Card.Subtitle>Back</Card.Subtitle>
						</Card.Header>
						<Card.Body style={{ height: "10em" }}>
							<Renderer data={props.data.back}></Renderer>
						</Card.Body>
					</Card>
				</animated.div>
			</div>
		</Container>
	)
}

PlayCardAnimated.whyDidYouRerender = true
export default PlayCardAnimated