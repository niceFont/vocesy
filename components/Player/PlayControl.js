import React, { useState } from "react"
import {Form,
	Row,
	Col,
	Container,
	ButtonGroup,
	Button,} from "react-bootstrap"


export const PlayControl = props => {
	const [userValue, setUserValue] = useState("")
	const [disabled, toggleDisabled] = useState(false)

	return (
		<Container
			style={{
				margin: "50px 0 100px 0px",
			}}
		>
			<Row className="justify-content-center">
				{props.done ? (
					<Col xs="6" sm="6" md="6">
						<Button block onClick={props.restart}>
              Restart
						</Button>
					</Col>
				) : (
					<React.Fragment>
						{props.settings.uv === "false" ? (
							<Col
								style={{
									marginTop: 350,
								}}
								xs="12"
								sm="10"
								lg="6"
								md="8"
							>
								<Form.Control
									placeholder="Enter your Answer..."
									style={{
										resize: "none",
									}}
									as="textarea"
									onKeyPress={e => {
										if (e.key === "Enter" || e.keyCode === 13) {
											if (e.target.value.trim() !== "") {
												e.preventDefault()
												e.target.value = "".replace(/(\r\n|\n|\r)/gm, "")
												if (props.current + 1 === props.max) {
													props.toggleDone(true)
												} else {
													props.next(curr => curr + 1)
												}
												props.pushInput(arr => [...arr, userValue])
												setUserValue("")
												return false
											}
										}
									}}
									onChange={({ target }) => setUserValue(target.value)}
								></Form.Control>
							</Col>
						) : (
							<Col
								style={{
									marginTop: 350,
								}}
								md="4"
								className="text-center"
							>
								<ButtonGroup>
									<Button
										disabled={props.done || disabled}
										variant="danger"
										onClick={() => {
											toggleDisabled(true)
											if (props.current + 1 === props.max) {
												props.toggleDone(done => !done)
											} else {
												props.next(curr => curr + 1)
											}
											props.pushInput(arr => [...arr, false])
											toggleDisabled(false)
										}}
									>
                    Wrong
									</Button>
									<Button
										disabled={props.done || disabled}
										variant="success"
										onClick={() => {
											toggleDisabled(true)
											if (props.current + 1 === props.max) {
												props.toggleDone(done => !done)
											} else {
												props.next(curr => curr + 1)
											}
											props.pushInput(arr => [...arr, true])
											toggleDisabled(false)
										}}
									>
                    Right
									</Button>
								</ButtonGroup>
							</Col>
						)}
					</React.Fragment>
				)}
			</Row>
		</Container>
	)
}