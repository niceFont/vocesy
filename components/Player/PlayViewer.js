import { Container, Row, Col, Card, Table } from "react-bootstrap"
import { PlayCard } from "./PlayCard"
import {faInfoCircle} from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon} from "@fortawesome/react-fontawesome"
export const PlayViewer = props => {

	function _setDiffs(diffs) {

		return {
			__html: diffs.inputDiff + "<span style='color: red'>" + diffs.lengthDiff.join("") + "</span>"
		}
	}


	return (
		<Container>
			<Row
				style={{
					margin: 20
				}}
				className="justify-content-center">
				<Col md="2" className="text-center">
					{!props.done ? 
						<span>{props.current + "/" + props.max}</span>
						:
						<span>Done</span>
					}
				</Col>
			</Row>
			{props.done ? (
				<Row>
					<Col className="text-center">
						<Table>
							<thead>
								<tr>
									<th>Number</th>
									<th>Result</th>
									<th>Answer</th>
									{props.settings.uv === "false" &&
									<th>Diffs</th>
									}
								</tr>
							</thead>
							<tbody>

								{props.verify().map((result, index) => {
									console.log(result)
									return (
										<tr key={index}>
											<td>{index + 1}</td>
											<td>{result.result ? "Right" : "Wrong"}</td>
											<td>{result.answer}</td>
											{props.settings.uv === "false" &&
											<td dangerouslySetInnerHTML={_setDiffs(result.diffs)}></td>
											}
										</tr>
									)
								})}
							</tbody>
						</Table>
					</Col>
				</Row>
			) : (
				<Row className="justify-content-center">
					<Col md="6" lg="4">
						{props.settings.uv === "false" ?
							<Card style={{
								height: "25em"
							}}>
								<Card.Header>Front</Card.Header>
								<Card.Body>
									<Card.Text>
										{props.data.front}
									</Card.Text>
								</Card.Body>
							</Card>
							:
							<div className="text-center">
								<span style={{
									color: "gray"
								}} ><FontAwesomeIcon icon={faInfoCircle} ></FontAwesomeIcon> Click the Card to see if you're right!</span>	
								<PlayCard data={props.data}></PlayCard>
							</div>
						}
					</Col>
				</Row>
			)}
		</Container>
	)
}
