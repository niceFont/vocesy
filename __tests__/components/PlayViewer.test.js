
/*eslint no-undef: off*/
/*eslint camelcase: off*/

import React from "react"
import { PlayViewer } from "../../components/Player/PlayViewer"
import renderer from "react-test-renderer"
import {mount} from "enzyme"

describe("PlayViewer", () => {
	it("should render correctly with passed Props.", () => {

		const tree = renderer.create(<PlayViewer
			flipped={false}
			done={false}
			max={1}
			current={1}
			settings={{
				uv: "true"
			}}
			data={[{
				front: "hello", back: "world", id: 1, deck_id: 1
			}]}
		></PlayViewer>)
        
		expect(tree.toJSON()).toMatchSnapshot()
	})
    
	it("should show Card Back on Click", () => {
		let toggleFlipped = jest.fn()

		const component = mount(<PlayViewer
			flipped={false}
			toggleFlipped={toggleFlipped}
			done={false}
			max={1}
			current={1}
			settings={{
				uv: "true"
			}}
			data={[{
				front: "hello", back: "world", id: 1, deck_id: 1
			}]}
		></PlayViewer>)

        
		component.find("div#toggler").simulate("click")

		expect(toggleFlipped).toHaveBeenCalledTimes(1)
        
		component.setProps({
			flipped: true
		})
        
	})
	

})
