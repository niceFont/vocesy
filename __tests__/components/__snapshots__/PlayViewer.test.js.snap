// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayViewer should render correctly with passed Props. 1`] = `
<div
  className="container"
>
  <div
    className="justify-content-center row"
    style={
      Object {
        "margin": 20,
      }
    }
  >
    <div
      className="text-center col-md-6"
    >
      <span>
        1/1
      </span>
    </div>
  </div>
  <div
    className="justify-content-center row"
  >
    <div
      className="text-center"
    >
      <span
        style={
          Object {
            "color": "gray",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-info-circle fa-w-16 "
          data-icon="info-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         Click the Card to see if you're right!
      </span>
      <div
        style={
          Object {
            "opacity": 0,
            "transform": "translate3d(100%,0,0)",
          }
        }
      >
        <div
          className="container"
          id="toggler"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "overflow": "hidden",
              "padding": "25px 0 25px 0",
              "position": "absolute",
            }
          }
        >
          <div>
            <div
              className="justify-content-center text-center"
              id="front"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "opacity": 1,
                  "transform": "perspective(600px) rotateY(0deg)",
                }
              }
            >
              <div
                className="mx-auto card"
                style={
                  Object {
                    "height": "22rem",
                    "position": "absolute",
                    "width": "16em",
                  }
                }
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-subtitle h6"
                  >
                    Front
                  </div>
                </div>
                <div
                  className="card-body"
                  style={
                    Object {
                      "height": "15em",
                    }
                  }
                >
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "hello",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div
              id="back"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "opacity": 0,
                  "transform": "perspective(600px) rotateY(0deg) rotateY(180deg)",
                }
              }
            >
              <div
                className="card"
                style={
                  Object {
                    "height": "22rem",
                    "width": "16em",
                  }
                }
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-subtitle h6"
                  >
                    Back
                  </div>
                </div>
                <div
                  className="card-body"
                  style={
                    Object {
                      "height": "10em",
                    }
                  }
                >
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "world",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
